Fix pasting files, default `image` file name, add type to meta, file type refactor

arturi opened the pull request:
Fixes #394
Fixes #389

*Note: file pasting is [experimental](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData), only support images, and file name string is always empty, because it doesn’t pass a reference to a file, but copies its data/blob instead, it seems.*

* add checks for empty `fullFileName` and `fileExtension`
* dashboard: if blob is empty on paste, log errors and return, skip `addFile`
* add default `duration` for `info` messages

arturi commented:
Maybe while we are here, I should make `getFileNameAndExtension` return an object: `{name: 'foo', ext: 'bar'`?

goto-bus-stop commented on src/core/Core.js:
```diff
@@ -867,6 +867,7 @@ class Uppy {
 
   info (message, type, duration) {
     const isComplexMessage = typeof message === 'object'
+    duration = typeof duration === 'undefined' ? 3000 : duration
```
how about a default parameter instead?

arturi replied:
Good idea, done, thx!

goto-bus-stop commented on src/core/Utils.js:
```diff
@@ -182,8 +177,14 @@ function getFileTypeExtension (mimeType) {
+* @return {array} [fileName, fileExt]
+*/
 function getFileNameAndExtension (fullFileName) {
+  if (!fullFileName) return
```
is this still necessary now that we check the `file.name` before calling this function? if this sometimes returns an array and sometimes returns `undefined`, we may have to check the result type whenever we use it.

arturi replied:
Right, gonna change that.

goto-bus-stop commented:
also yeah i like the object idea 👍

arturi commented:
> maybe we should also assign a default name to file.meta.name

good idea, what worries me here though is, when I paste `.jpg` it gets a type of `image/png`. I guess that’s because it’s converted to `png`, so it’s cool?

arturi commented:
Updated:

- Added “The mime type could make sense to add by default too” from #389
- Changed the [name assignment](https://github.com/transloadit/uppy/blob/4b21fb7c83d41f21d76a6f99bc3f052c0fc2053f/src/core/Core.js#L309-L316)
- Moved from `file.type.general`, `file.type.specific` to `file.type = 'image/jpeg'`

goto-bus-stop commented on src/plugins/Dashboard/getFileTypeIcon.js:
```diff
@@ -1,6 +1,16 @@
 
-module.exports = function getIconByMime (fileTypeGeneral, fileTypeSpecific) {
+module.exports = function getIconByMime (fileType) {
+  const defaultChoise = {
```
very minor but `Choise` → `Choice`

arturi replied:
👍
